service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/PlannedOccupancyTest.kt

```diff
@@ -326,6 +422,83 @@ class PlannedOccupancyTest : FullApplicationTest() {
             listOf(OccupancyPeriod(defaultPeriod, 0.0, 0)),
             result
         )
+
+        val reportResult1 = fetchAndParsePlannedOccupancyReportForUnitDay(testAreaId, testDaycare.id, defaultPeriod.start)
+        assertEquals(0.0, reportResult1?.sum)
+        assertEquals(0, reportResult1?.headcount)
+        val reportResult2 = fetchAndParsePlannedOccupancyReportForUnitDay(testAreaId, testDaycare.id, defaultPeriod.end)
+        assertEquals(0.0, reportResult2?.sum)
+        assertEquals(0, reportResult2?.headcount)
+    }
+
+    @Test
+    fun `when child is planned to transfer to another unit the child is counted into both units`() {
+        val childId = UUID.randomUUID()
+        val daycareId1 = testDaycare.id
+        val daycareId2 = UUID.randomUUID()
+
+        db.transaction { tx ->
+            tx.handle.insertTestPerson(DevPerson(id = childId, dateOfBirth = LocalDate.of(2015, 1, 1)))
+            tx.handle.insertTestChild(DevChild(id = childId))
+            tx.handle.insertTestDaycare(DevDaycare(id = daycareId2, areaId = testDaycare.areaId, name = "foo"))
+
+            tx.handle.insertTestPlacement(DevPlacement(type = PlacementType.DAYCARE, childId = childId, unitId = daycareId1, startDate = defaultPeriod.start, endDate = defaultPeriod.end))
+
+            val applicationId = insertTestApplication(tx.handle, childId = childId)
+            insertTestPlacementPlan(h = tx.handle, applicationId = applicationId, unitId = daycareId2, type = PlacementType.DAYCARE_PART_TIME, startDate = period2.start, endDate = period2.end)
+        }
+
+        val result1 = fetchAndParseOccupancy(daycareId1, defaultPeriod)
+        assertEquals(
+            listOf(OccupancyPeriod(defaultPeriod, 1.0, 1)),
+            result1
+        )
+
+        val result2 = fetchAndParseOccupancy(daycareId2, defaultPeriod)
+        assertEquals(
+            listOf(
+                OccupancyPeriod(period1, 0.0, 0),
+                OccupancyPeriod(period2, 0.54, 1)
+            ),
+            result2
+        )
+
+        val reportResult1 = fetchAndParsePlannedOccupancyReportForUnitDay(testAreaId, daycareId1, period2.start)
+        assertEquals(1.0, reportResult1?.sum)
+        assertEquals(1, reportResult1?.headcount)
+
+        val reportResult2 = fetchAndParsePlannedOccupancyReportForUnitDay(testAreaId, daycareId2, period2.start)
+        assertEquals(0.54, reportResult2?.sum)
+        assertEquals(1, reportResult2?.headcount)
+    }
+
+    @Test
+    fun `when child is planned to transfer to same unit the higher capacity wins`() {
```

Comment:
I think a test where the situation is opposite (planned placement has higher occupancy) would be valuable as well.

Reply:
hmm actually the tests `occupancy calculation is correct for child over 3 years old with full time placement plan over an existing part time placement`  and `occupancy calculation is correct for child over 3 years old with part time placement plan over an existing full time placement`  already cover both ways so this test can be deleted as duplicate?

Reply:
I guess so, but the assertion in `occupancy calculation is correct for child over 3 years old with part time placement plan over an existing full time placement` expects the opposite of what you intend (I think) 😄

Reply:
it used to, but I changed the assertion (line 351)

Reply:
Oh yeah, sorry, I was looking at master.